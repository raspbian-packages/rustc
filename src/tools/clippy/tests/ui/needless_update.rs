#![feature(plugin)]
#![plugin(clippy)]

#![warn(needless_update)]
#![allow(no_effect)]

struct S {
    pub a: i32,
    pub b: i32,
}

fn main() {
    let base = S { a: 0, b: 0 };
    S { ..base }; // no error
    S { a: 1, ..base }; // no error
    S { a: 1, b: 1, ..base };
}
