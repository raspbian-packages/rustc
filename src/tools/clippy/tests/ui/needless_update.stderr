error: struct update has no effect, all the fields in the struct have already been specified
  --> $DIR/needless_update.rs:16:23
   |
16 |     S { a: 1, b: 1, ..base };
   |                       ^^^^
   |
   = note: `-D needless-update` implied by `-D warnings`

error: aborting due to previous error

